### PR TITLE
Use URLSearchParams for parsing search params 

### PR DESCRIPTION
**Pull request by mohsen1**
URL search params is _a lot_ more involved than simple split by `&` and making a a key/value pair from it. I noticed in the router an assumption is made that search params are not repeated. This PR does not fix the issue because this assumption goes deeper in the router. Instead, I've introduced standard `URLSearchParasm` for parsing the search params which handles a lot of corner cases such as unicode and also allow us to get a list for repeated search arguments if we decided to support that. 

I've also updated comments in the util.js to use JSDoc so comments show up in hover in editors. 

Also used `String.prototype.startsWith` instead of `path[0]` for better readability.

<hr>
P.S. This is my first PR and you should expect a lot more contribution from me to come! I love how this project is set up and the decisions that are made are almost perfect! ❤️

### CODE REVIEW

File: packages/router/src/util.js [hunk 1]
```diff
@@ -73,33 +77,35 @@ const matchPath = (route, pathname, paramTypes) => {
...
-  pairs.forEach((pair) => {
-    const keyval = pair.split('=')
-    searchProps[keyval[0]] = keyval[1] || ''
-  })
-  return searchProps
+  const searchParams = new URLSearchParams(search);
+
+  return [...searchParams.keys()].reduce((params, key) => ({
+    ...params,
+    [key]: searchParams.get(key)
```
Comment (mohsen1) [inline packages/router/src/util.js]: Here it is possible to do `searchParams.getAll(key)` instead

**Comment by mojombo** [comment]
Great suggestion! I wonder if there's a way to have Babel dictate the inclusion of the polyfill instead of including it directly. We'd like to end up allowing users to specify what browser support they want to satisfy and have all Redwood dependencies be transpiled for that specification, at which point Babel could make the decision of whether this polyfill was needed or not.

@peterp curious what our latest thinking is on this.

**Comment by mohsen1** [comment]
So according to [Babel docs for preset-env](https://babeljs.io/docs/en/babel-preset-env) we can enable `useBuiltIns` and set it to "usage", then `core-js` will be pollyfilling things like `URLSearchParams` according to the browserlist config. 

Going to try this out and let you know how it works in practice.

**Comment by mohsen1** [comment]
Ok, if Redwood is compiling router in userland using Babel, it will add the polyfill as it does in compiled version here: 

<details>

```js
"use strict";

var _interopRequireDefault = require("@babel/runtime-corejs3/helpers/interopRequireDefault");

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports.replaceParams = exports.validatePath = exports.parseSearch = exports.matchPath = exports.createNamedContext = void 0;

var _concat = _interopRequireDefault(require("@babel/runtime-corejs3/core-js/instance/concat"));

var _keys = _interopRequireDefault(require("@babel/runtime-corejs3/core-js/object/keys"));

var _forEach = _interopRequireDefault(require("@babel/runtime-corejs3/core-js/instance/for-each"));

var _getIterator2 = _interopRequireDefault(require("@babel/runtime-corejs3/core-js/get-iterator"));

var _startsWith = _interopRequireDefault(require("@babel/runtime-corejs3/core-js/instance/starts-with"));

var _keys2 = _interopRequireDefault(require("@babel/runtime-corejs3/core-js/instance/keys"));

var _urlSearchParams = _interopRequireDefault(require("@babel/runtime-corejs3/core-js/url-search-params"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/defineProperty"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/slicedToArray"));

var _slice = _interopRequireDefault(require("@babel/runtime-corejs3/core-js/instance/slice"));
```

</details>

**Comment by thedavidprice** [comment]
Our CI Checks were not running for PRs from external, forked repos. See #358 

This is now fixed but will not retroactively apply to this PR. Before merging, please locally confirm the following checks pass:
- [ ] `yarn build`
- [ ] `yarn lint`
- [ ] `yarn test`

Thanks in advance!

**Pull request merged**